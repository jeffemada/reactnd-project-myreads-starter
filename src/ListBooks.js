import React from 'react';
import PropTypes from 'prop-types';
import Book from './Book';

function ListBooks(props) {
  const { books, onRefresh } = props;

  return (
    <ol className="books-grid">
      {books.map((book) => (
        <Book key={book.id} book={book} onRefresh={onRefresh} />
      ))}
    </ol>
  );
}

ListBooks.propTypes = {
  books: PropTypes.array.isRequired,
  onRefresh: PropTypes.func.isRequired
};

export default ListBooks;
