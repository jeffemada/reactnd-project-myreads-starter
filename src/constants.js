export const SHELVES = [
  { id: "wantToRead", title: "Want to Read" },
  { id: "currentlyReading", title: "Currently Reading" },
  { id: "read", title: "Read" },
  { id: "none", title: "None" }
];
