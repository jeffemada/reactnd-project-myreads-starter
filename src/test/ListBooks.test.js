import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import ListBooks from '../ListBooks';
import { MY_BOOKS_MOCK } from './constants';

Enzyme.configure({ adapter: new Adapter() });

it('should render a ListBooks', () => {
  const wrapper = shallow(<ListBooks books={MY_BOOKS_MOCK} onRefresh={jest.fn()} onLoading={jest.fn()} />);
  expect(wrapper).toMatchSnapshot();
});
