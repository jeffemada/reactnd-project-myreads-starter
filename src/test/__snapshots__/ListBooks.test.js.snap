// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a ListBooks 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListBooks
    books={
      Array [
        Object {
          "authors": Array [
            "Divya Manian",
          ],
          "id": "aaHXYmZLYNQC",
          "imageLinks": Object {
            "smallThumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
            "thumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
          },
          "shelf": "wantToRead",
          "title": "HTML5 Boilerplate Web Development",
        },
        Object {
          "authors": Array [
            "Julie Ratner",
          ],
          "id": "sKvi5fVL-uAC",
          "imageLinks": Object {
            "smallThumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
            "thumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
          },
          "shelf": "read",
          "title": "Human Factors and Web Development",
        },
        Object {
          "authors": Array [
            "Cassio de Sousa Antonio",
          ],
          "id": "PKpPCwAAQBAJ",
          "imageLinks": Object {
            "thumbnail": "http://books.google.com/books/content?id=PKpPCwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
          },
          "shelf": "wantToRead",
          "title": "Pro React",
        },
      ]
    }
    onLoading={[MockFunction]}
    onRefresh={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="list-books-title"
        >
          <h1>
            MyReads
          </h1>
        </div>,
        <div
          className="list-books-content"
        >
          <Bookshelf
            books={
              Array [
                Object {
                  "authors": Array [
                    "Divya Manian",
                  ],
                  "id": "aaHXYmZLYNQC",
                  "imageLinks": Object {
                    "smallThumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                    "thumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                  },
                  "shelf": "wantToRead",
                  "title": "HTML5 Boilerplate Web Development",
                },
                Object {
                  "authors": Array [
                    "Cassio de Sousa Antonio",
                  ],
                  "id": "PKpPCwAAQBAJ",
                  "imageLinks": Object {
                    "thumbnail": "http://books.google.com/books/content?id=PKpPCwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                  },
                  "shelf": "wantToRead",
                  "title": "Pro React",
                },
              ]
            }
            onLoading={[MockFunction]}
            onRefresh={[MockFunction]}
            shelf={
              Object {
                "id": "wantToRead",
                "title": "Want to Read",
              }
            }
          />
          <Bookshelf
            books={Array []}
            onLoading={[MockFunction]}
            onRefresh={[MockFunction]}
            shelf={
              Object {
                "id": "currentlyReading",
                "title": "Currently Reading",
              }
            }
          />
          <Bookshelf
            books={
              Array [
                Object {
                  "authors": Array [
                    "Julie Ratner",
                  ],
                  "id": "sKvi5fVL-uAC",
                  "imageLinks": Object {
                    "smallThumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                    "thumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                  },
                  "shelf": "read",
                  "title": "Human Factors and Web Development",
                },
              ]
            }
            onLoading={[MockFunction]}
            onRefresh={[MockFunction]}
            shelf={
              Object {
                "id": "read",
                "title": "Read",
              }
            }
          />
        </div>,
        <AddBook />,
      ],
      "className": "list-books",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            MyReads
          </h1>,
          "className": "list-books-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "MyReads",
          },
          "ref": null,
          "rendered": "MyReads",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Bookshelf
              books={
                Array [
                  Object {
                    "authors": Array [
                      "Divya Manian",
                    ],
                    "id": "aaHXYmZLYNQC",
                    "imageLinks": Object {
                      "smallThumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                      "thumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                    },
                    "shelf": "wantToRead",
                    "title": "HTML5 Boilerplate Web Development",
                  },
                  Object {
                    "authors": Array [
                      "Cassio de Sousa Antonio",
                    ],
                    "id": "PKpPCwAAQBAJ",
                    "imageLinks": Object {
                      "thumbnail": "http://books.google.com/books/content?id=PKpPCwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                    },
                    "shelf": "wantToRead",
                    "title": "Pro React",
                  },
                ]
              }
              onLoading={[MockFunction]}
              onRefresh={[MockFunction]}
              shelf={
                Object {
                  "id": "wantToRead",
                  "title": "Want to Read",
                }
              }
            />,
            <Bookshelf
              books={Array []}
              onLoading={[MockFunction]}
              onRefresh={[MockFunction]}
              shelf={
                Object {
                  "id": "currentlyReading",
                  "title": "Currently Reading",
                }
              }
            />,
            <Bookshelf
              books={
                Array [
                  Object {
                    "authors": Array [
                      "Julie Ratner",
                    ],
                    "id": "sKvi5fVL-uAC",
                    "imageLinks": Object {
                      "smallThumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                      "thumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                    },
                    "shelf": "read",
                    "title": "Human Factors and Web Development",
                  },
                ]
              }
              onLoading={[MockFunction]}
              onRefresh={[MockFunction]}
              shelf={
                Object {
                  "id": "read",
                  "title": "Read",
                }
              }
            />,
          ],
          "className": "list-books-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "wantToRead",
            "nodeType": "function",
            "props": Object {
              "books": Array [
                Object {
                  "authors": Array [
                    "Divya Manian",
                  ],
                  "id": "aaHXYmZLYNQC",
                  "imageLinks": Object {
                    "smallThumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                    "thumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                  },
                  "shelf": "wantToRead",
                  "title": "HTML5 Boilerplate Web Development",
                },
                Object {
                  "authors": Array [
                    "Cassio de Sousa Antonio",
                  ],
                  "id": "PKpPCwAAQBAJ",
                  "imageLinks": Object {
                    "thumbnail": "http://books.google.com/books/content?id=PKpPCwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                  },
                  "shelf": "wantToRead",
                  "title": "Pro React",
                },
              ],
              "onLoading": [MockFunction],
              "onRefresh": [MockFunction],
              "shelf": Object {
                "id": "wantToRead",
                "title": "Want to Read",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "currentlyReading",
            "nodeType": "function",
            "props": Object {
              "books": Array [],
              "onLoading": [MockFunction],
              "onRefresh": [MockFunction],
              "shelf": Object {
                "id": "currentlyReading",
                "title": "Currently Reading",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "read",
            "nodeType": "function",
            "props": Object {
              "books": Array [
                Object {
                  "authors": Array [
                    "Julie Ratner",
                  ],
                  "id": "sKvi5fVL-uAC",
                  "imageLinks": Object {
                    "smallThumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                    "thumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                  },
                  "shelf": "read",
                  "title": "Human Factors and Web Development",
                },
              ],
              "onLoading": [MockFunction],
              "onRefresh": [MockFunction],
              "shelf": Object {
                "id": "read",
                "title": "Read",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="list-books-title"
          >
            <h1>
              MyReads
            </h1>
          </div>,
          <div
            className="list-books-content"
          >
            <Bookshelf
              books={
                Array [
                  Object {
                    "authors": Array [
                      "Divya Manian",
                    ],
                    "id": "aaHXYmZLYNQC",
                    "imageLinks": Object {
                      "smallThumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                      "thumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                    },
                    "shelf": "wantToRead",
                    "title": "HTML5 Boilerplate Web Development",
                  },
                  Object {
                    "authors": Array [
                      "Cassio de Sousa Antonio",
                    ],
                    "id": "PKpPCwAAQBAJ",
                    "imageLinks": Object {
                      "thumbnail": "http://books.google.com/books/content?id=PKpPCwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                    },
                    "shelf": "wantToRead",
                    "title": "Pro React",
                  },
                ]
              }
              onLoading={[MockFunction]}
              onRefresh={[MockFunction]}
              shelf={
                Object {
                  "id": "wantToRead",
                  "title": "Want to Read",
                }
              }
            />
            <Bookshelf
              books={Array []}
              onLoading={[MockFunction]}
              onRefresh={[MockFunction]}
              shelf={
                Object {
                  "id": "currentlyReading",
                  "title": "Currently Reading",
                }
              }
            />
            <Bookshelf
              books={
                Array [
                  Object {
                    "authors": Array [
                      "Julie Ratner",
                    ],
                    "id": "sKvi5fVL-uAC",
                    "imageLinks": Object {
                      "smallThumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                      "thumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                    },
                    "shelf": "read",
                    "title": "Human Factors and Web Development",
                  },
                ]
              }
              onLoading={[MockFunction]}
              onRefresh={[MockFunction]}
              shelf={
                Object {
                  "id": "read",
                  "title": "Read",
                }
              }
            />
          </div>,
          <AddBook />,
        ],
        "className": "list-books",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              MyReads
            </h1>,
            "className": "list-books-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MyReads",
            },
            "ref": null,
            "rendered": "MyReads",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Bookshelf
                books={
                  Array [
                    Object {
                      "authors": Array [
                        "Divya Manian",
                      ],
                      "id": "aaHXYmZLYNQC",
                      "imageLinks": Object {
                        "smallThumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                        "thumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                      },
                      "shelf": "wantToRead",
                      "title": "HTML5 Boilerplate Web Development",
                    },
                    Object {
                      "authors": Array [
                        "Cassio de Sousa Antonio",
                      ],
                      "id": "PKpPCwAAQBAJ",
                      "imageLinks": Object {
                        "thumbnail": "http://books.google.com/books/content?id=PKpPCwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                      },
                      "shelf": "wantToRead",
                      "title": "Pro React",
                    },
                  ]
                }
                onLoading={[MockFunction]}
                onRefresh={[MockFunction]}
                shelf={
                  Object {
                    "id": "wantToRead",
                    "title": "Want to Read",
                  }
                }
              />,
              <Bookshelf
                books={Array []}
                onLoading={[MockFunction]}
                onRefresh={[MockFunction]}
                shelf={
                  Object {
                    "id": "currentlyReading",
                    "title": "Currently Reading",
                  }
                }
              />,
              <Bookshelf
                books={
                  Array [
                    Object {
                      "authors": Array [
                        "Julie Ratner",
                      ],
                      "id": "sKvi5fVL-uAC",
                      "imageLinks": Object {
                        "smallThumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                        "thumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                      },
                      "shelf": "read",
                      "title": "Human Factors and Web Development",
                    },
                  ]
                }
                onLoading={[MockFunction]}
                onRefresh={[MockFunction]}
                shelf={
                  Object {
                    "id": "read",
                    "title": "Read",
                  }
                }
              />,
            ],
            "className": "list-books-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "wantToRead",
              "nodeType": "function",
              "props": Object {
                "books": Array [
                  Object {
                    "authors": Array [
                      "Divya Manian",
                    ],
                    "id": "aaHXYmZLYNQC",
                    "imageLinks": Object {
                      "smallThumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                      "thumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                    },
                    "shelf": "wantToRead",
                    "title": "HTML5 Boilerplate Web Development",
                  },
                  Object {
                    "authors": Array [
                      "Cassio de Sousa Antonio",
                    ],
                    "id": "PKpPCwAAQBAJ",
                    "imageLinks": Object {
                      "thumbnail": "http://books.google.com/books/content?id=PKpPCwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                    },
                    "shelf": "wantToRead",
                    "title": "Pro React",
                  },
                ],
                "onLoading": [MockFunction],
                "onRefresh": [MockFunction],
                "shelf": Object {
                  "id": "wantToRead",
                  "title": "Want to Read",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "currentlyReading",
              "nodeType": "function",
              "props": Object {
                "books": Array [],
                "onLoading": [MockFunction],
                "onRefresh": [MockFunction],
                "shelf": Object {
                  "id": "currentlyReading",
                  "title": "Currently Reading",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "read",
              "nodeType": "function",
              "props": Object {
                "books": Array [
                  Object {
                    "authors": Array [
                      "Julie Ratner",
                    ],
                    "id": "sKvi5fVL-uAC",
                    "imageLinks": Object {
                      "smallThumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                      "thumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                    },
                    "shelf": "read",
                    "title": "Human Factors and Web Development",
                  },
                ],
                "onLoading": [MockFunction],
                "onRefresh": [MockFunction],
                "shelf": Object {
                  "id": "read",
                  "title": "Read",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
